src/ProjectService.Data/ProjectRepository.cs

```diff
@@ -49,21 +50,20 @@ public void CreateNewProject(DbProject newProject)
             provider.Save();
         }
 
-        public Guid EditProjectById(DbProject dbProject)
+        public bool EditProject(Guid projectId, JsonPatchDocument<DbProject> request)
         {
             var projectToEdit = provider.Projects
-                .AsNoTracking()
-                .FirstOrDefault(p => p.Id == dbProject.Id);
+                .FirstOrDefault(p => p.Id == projectId);
 
             if (projectToEdit == null)
             {
                 throw new NullReferenceException("Project with this Id does not exist");
```

Comment:
I think better is to write id of no exists project in the message instead 'this'

Reply:
done